src/main/java/fi/aalto/cs/apluscourses/intellij/model/task/IntelliJComponentPresenterBase.java

```diff
@@ -56,4 +73,33 @@ private void removeHighlightInternal() {
   }
 
   protected abstract GenericHighlighter getHighlighter();
+
+  protected abstract boolean tryToShow();
+
+  @Override
+  public boolean isVisible() {
+    var highlighter = getHighlighter();
+    return highlighter != null && highlighter.getComponent().isVisible();
+  }
+
+  @Override
+  public Timer getTimer() {
+    return timer;
+  }
+
+  private void startTimer() {
+    timer.scheduleAtFixedRate(new TaskRefresher(), REFRESH_INTERVAL, REFRESH_INTERVAL);
+  }
+
+  private class TaskRefresher extends TimerTask {
+
+    @Override
+    public void run() {
+      ApplicationManager.getApplication().invokeLater(() -> {
+        if (!isVisible()) {
+          highlight();
+        }
+      });
```

Comment:
Remark: This calls `highlight` which is also internally wrapped in `invokeLater`. Is there perhaps some unnecessary `invokeLater`s here. (Having nested `invokeLater`s is not a problem but usually a sign of unclear organization of code.)